src/rest.rs

```diff
@@ -1187,6 +1189,26 @@ fn json_response<T: Serialize>(value: T, ttl: u32) -> Result<Response<Body>, Htt
         .unwrap())
 }
 
+fn json_maybe_error_response<T: Serialize>(
+    value: Result<T, errors::Error>,
+    ttl: u32,
+) -> Result<Response<Body>, HttpError> {
+    let response = Response::builder()
+        .header("Content-Type", "application/json")
+        .header("Cache-Control", format!("public, max-age={:}", ttl));
+    Ok(match value {
+        Ok(v) => response
+            .body(Body::from(serde_json::to_string(&v)?))
+            .expect("Valid http response"),
+        Err(e) => response
```

Comment:
Perhaps we should lower Cache-Control max-age for the Err variant?

Give it more of a chance to re-check if it fixed itself.